salt/cloud/clouds/opennebula.py

```diff
@@ -901,17 +917,24 @@ def create(vm_):
         {'kwargs': kwargs},
     )
 
-    region = ''
+    template = []
     if kwargs['region_id'] is not None:
-        region = 'SCHED_REQUIREMENTS="ID={0}"'.format(kwargs['region_id'])
+        template.append('SCHED_REQUIREMENTS="ID={0}"'.format(kwargs['region_id']))
+    if vm_['memory'] is not None:
```

Comment:
Can you change this to a `vm_.get('memory')` call? That way we can avoid a KeyError if the vm_ passed in doesn't have this option and will default to `None`. The same comment goes for all of the options you've added below.